src/datasets/packaged_modules/folder_based_builder/folder_based_builder.py

```diff
@@ -58,6 +59,9 @@ def _split_generators(self, dl_manager):
         if not self.config.data_files:
             raise ValueError(f"At least one data file must be specified, but got data_files={self.config.data_files}")
 
+        if self.config.metadata_filename:
+            self.METADATA_FILENAMES = [self.config.metadata_filename]
```

Comment:
This would change `METADATA_FILENAMES` for all the instances of FolderBasedBuilder.

You should define one per instance in `__init__` no ?